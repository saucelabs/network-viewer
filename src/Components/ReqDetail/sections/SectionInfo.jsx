import React, { useState } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';

import Styles from './../ReqDetail.styles.scss';
import SectionTitle from './SectionTitle';

const context = classNames.bind(Styles);

const SectionInfo = ({
  eventKey,
  data,
  component,
  isEncodeEnabled,
  isParseEnabled,
  isVisible,
}) => {
  const [isOpen, setIsOpen] = useState(isVisible);
  const [isPayloadTransformed, updateTransform] = useState(true);

  const handlePayloadTransform = () => updateTransform(!isPayloadTransformed);
  const ChildComponent = () => component({
    data,
    isPayloadTransformed,
    onChangeEncode: handlePayloadTransform,
  });

  return (
    <>
      <SectionTitle
        eventKey={eventKey}
        isEncodeEnabled={isEncodeEnabled}
        isOpen={isOpen}
        isParseEnabled={isParseEnabled}
        isPayloadTransformed={isPayloadTransformed}
        onClick={() => setIsOpen(!isOpen)}
        onPayloadTransform={handlePayloadTransform}
      />
      {isOpen && <ChildComponent />}
    </>
  );
};

SectionInfo.propTypes = {
  component: PropTypes.func.isRequired,
  data: PropTypes.object,
  eventKey: PropTypes.string.isRequired,
  isEncodeEnabled: PropTypes.bool,
  isParseEnabled: PropTypes.bool,
  isVisible: PropTypes.bool,
};

SectionInfo.defaultProps = {
  data: null,
  isEncodeEnabled: false,
  isParseEnabled: false,
  isVisible: false,
};

export default SectionInfo;
