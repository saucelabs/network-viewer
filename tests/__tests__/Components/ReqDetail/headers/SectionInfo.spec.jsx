import React from 'react';
import { shallow } from 'enzyme';

import SectionInfo from '../../../../../src/Components/ReqDetail/sections/SectionInfo';

describe('SectionInfo', () => {
  const props = {
    eventKey: 'general',
    data: {},
    component: () => {},
  };

  it('renders without crashing', () => {
    const element = shallow(<SectionInfo {...props} />);
    expect(element).toMatchSnapshot();
  });
});
