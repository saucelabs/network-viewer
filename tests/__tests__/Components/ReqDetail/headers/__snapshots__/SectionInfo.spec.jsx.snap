// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionInfo renders without crashing 1`] = `
<Fragment>
  <SectionTitle
    eventKey="general"
    isEncodeEnabled={false}
    isOpen={false}
    isParseEnabled={false}
    isPayloadTransformed={true}
    onClick={[Function]}
    onPayloadTransform={[Function]}
  />
</Fragment>
`;
